src/libos/src/fs/disk_file.rs

```diff
@@ -131,6 +134,25 @@ impl DiskFile {
     pub fn ioctl(&self, _cmd: &mut dyn IoctlCmd) -> Result<()> {
         return_errno!(EINVAL, "this file does not support ioctl");
     }
+
+    pub fn metadata(&self) -> Metadata {
+        Metadata {
+            dev: 0,
+            inode: 0x1000,
+            size: self.disk.total_bytes(),
+            blk_size: BLOCK_SIZE,
+            blocks: self.disk.total_blocks(),
+            atime: Timespec { sec: 0, nsec: 0 },
+            mtime: Timespec { sec: 0, nsec: 0 },
+            ctime: Timespec { sec: 0, nsec: 0 },
+            type_: FileType::File,
```

Comment:
Should be `FileType::BlockDevice`. User programs (e.g., fio) may use `S_ISBLK` macro to test if a value of `struct stat` type refers to a block device or not.

Reply:
Actually, I have tried to use `BlockDevice`, but FIO will access it as a block device with some `ioctls`. As we already treat the block device as a file to read and write, it is reasonable to use `File`.